Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOptionType.h

```diff
@@ -0,0 +1,51 @@
+/*
+* Backpack - Skyscanner's Design System
+*
+* Copyright 2018-2020 Skyscanner Ltd
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Represents a single navigation option in `BPKHorizontalNavigation`.
+ * For common use cases `BPKHorizontalNavigationOption` is provided but
+ * by implementing a custom type that conform to this protocol you can
+ * implement custom navigation items.
+ *
+ * @see BPKHorizontalNavigationItem
+ * @see BPKHorizontalNavigationOption
+ */
+@protocol BPKHorizontalNavigationOptionType <NSObject>
+@required
+
+/**
+ * A tag representing the option, can be used to distinguish options from
+ * eachother.
+*/
+@property(nonatomic, readonly) NSInteger tag;
+
+/**
+ * A `UIControl` sublclass to use for rendering this item
+ * in a `BPKHorizontalNavigation`. In addition to being a `UIControl`
+ * subclass it should implement the initializer
+ *
+ *      - (instancetype)initWithOption:(id<BPKHorizontalNavigationOptionType>)option;
+ */
+@property(class, nonatomic, readonly) Class itemClass;
```

Comment:
Can we make this a `UIControl` instead of `Class` to ensure that at least is inheriting from `UIControl`?
In addition, we can move the initializer `- (instancetype)initWithOption:(BPKHorizontalNavigationOption *)option` to a protocol to force this property to be a `UIControl<someProtocolWithTheInitializer>`

Reply:
That doens't work because it's not returning an instance it's just returning a `Class` which we later intialize. I don't believe there's any way to constrain the type of a `Class`

Reply:
But maybe it's just cleaner to return an instance rather than a class here anyway